bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java

```diff
@@ -118,6 +120,20 @@ protected AbstractConfiguration() {
         }
     }
 
+    /**
+     * Limit who can start the application to prevent future permission errors.
+     */
+    public void setPermittedStartupUsers(String s) {
+        setProperty(PERMITTED_STARTUP_USERS, s);
+    }
+
+    /**
+     * Get array of users specified in this property.
+     */
+    public String[] getPermittedStartupUsers() {
```

Comment:
Can't we have consistent return type, in regards to the setter?

Reply:
@eolivelli Well, the reason for the asymmetry is that org.apache.commons.configuration.AbstractConfiguration.setProperty has magic to split strings containing the delimiter (',' by default) into a list prior to storing internally.  getStringArray there simply either returns the stored array or, if it's a string, returns it as a single element array.  I could instead choose to make setPermittedStartupUsers take an array, but it would make the unit tests (which are the only user) less useful since it would bypass the string splitting behavior.  If there's ever another user, it would be perfectly ok to add a variant that takes a String[] as well, but I don't see a reason to do it now.